python/beeai_framework/agents/experimental/agent.py

```diff
@@ -264,20 +277,9 @@ async def handler(run_context: RunContext) -> RequirementAgentRunOutput[TOutput]
 
                 await self.memory.add_many(extract_last_tool_call_pair(state.memory) or [])
 
-            return RequirementAgentRunOutput(
-                answer=state.answer, memory=state.memory, state=state, answer_structured=state.result
-            )
+            return RequirementAgentOutput(output=[state.answer], output_structured=state.result, state=state)
 
-        return self._to_run(
-            handler,
-            signal=signal,
-            run_params={
-                "prompt": prompt,
-                "context": context,
-                "expected_output": expected_output,
-                "execution": execution,
-            },
-        )
+        return await handler(RunContext.get())
```

Comment:
We should remove the handler wrapper in a follow-up PR, but I assume you did it this way to avoid too many changes at once and to keep the diff clear 👍🏻